src/Storages/Statistics/StatisticsMinMax.cpp

```diff
@@ -0,0 +1,87 @@
+#include <Storages/Statistics/StatisticsMinMax.h>
+#include <Common/FieldVisitorConvertToNumber.h>
+#include <DataTypes/DataTypeLowCardinality.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+#include <Interpreters/convertFieldToType.h>
+
+#include <algorithm>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int ILLEGAL_STATISTICS;
+}
+
+StatisticsMinMax::StatisticsMinMax(const SingleStatisticsDescription & statistics_description, const DataTypePtr & data_type_)
+    : IStatistics(statistics_description)
+    , data_type(data_type_)
+{
+}
+
+void StatisticsMinMax::update(const ColumnPtr & column)
+{
+    for (size_t row = 0; row < column->size(); ++row)
+    {
+        if (column->isNullAt(row))
+            continue;
+
+        auto value = column->getFloat64(row);
+        min = std::min(value, min);
+        max = std::max(value, max);
+    }
+    row_count += column->size();
+}
+
+void StatisticsMinMax::serialize(WriteBuffer & buf)
+{
+    writeIntBinary(row_count, buf);
+    writeFloatBinary(min, buf);
+    writeFloatBinary(max, buf);
+}
+
+void StatisticsMinMax::deserialize(ReadBuffer & buf)
+{
+    readIntBinary(row_count, buf);
+    readFloatBinary(min, buf);
+    readFloatBinary(max, buf);
+}
+
+Float64 StatisticsMinMax::estimateLess(const Field & val) const
+{
+    Field val_converted = convertFieldToType(val, *data_type);
```

Comment:
Why not convert to Float64 directly here? What will happen if it is `val < 10.6` but `val` is Int ?

Reply:
I guess the `val ` you mentioned here is a column.  Let's assume the expression `col_date` > '2024-08-07', if we convert `2024-08-07` directly to Float64, we will get `Null`, so I converted it to the real column data type and then translate it to Float64.

But you provided an edge case, for expression `col_int32 < 10.6`, we will also get `Null` which is not consistent with the real result.  Thanks, I will deal with it.